apps/full-site-editing/full-site-editing-plugin/starter-page-templates/page-template-modal/components/block-preview.js

```diff
@@ -13,6 +13,22 @@ import { BlockEditorProvider, BlockList } from '@wordpress/block-editor';
 import { Disabled } from '@wordpress/components';
 import { PostTitle } from '@wordpress/editor';
 
+/**
+ * Returns true if current page is a front (home) page and the
+ * "hide_front_page_title" theme mod is on.
+ *
+ * @returns {boolean} homepage title visibility flag
+ */
+function hideHomepageTitle() {
+	if ( ! window.starterPageTemplatesConfig ) {
+		return false;
+	}
+
+	const { isFrontPage, hideFrontPageTitle } = window.starterPageTemplatesConfig;
```

Comment:
Can we try to pass the those as props from the top component or better change the component in this file to be "less clever" and only have a boolean prop like `hideHomepageTitle` that gets passed from the parent? 

We've tried to avoid touching `starterPageTemplatesConfig` from anywhere other than the single file with plugin initialization just so we can avoid so many checks like you do here for the presence of the object. See the source code:

https://github.com/Automattic/wp-calypso/search?q=starterPageTemplatesConfig&unscoped_q=starterPageTemplatesConfig